File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
#include <numeric>
using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
        set<int> num;
    int max = 0;
    for (auto e : elements)
    {
        max += e;
        num.insert(e);
    }
    num.insert(max);
    for (int i = 1; i < elements.size(); i++)
    {
        auto prev = elements.begin();
        auto next = elements.begin() + i;
        int sum = accumulate(prev, next+1, 0);
        for (int j = 0;j< elements.size(); j++)
        {
            

            next++;
            if (next == elements.end())
                next = elements.begin();
            sum += *next;
            sum -= *prev;
            prev++;
            num.insert(sum);
        }
    }
    answer = num.size();
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        HashSet<int> sum = new HashSet<int>();
        int size = elements.Length;
        
        for(int i = 0; i< size;i++)
        {
            int all = 0;
            for(int j = i; j<i+size;j++)
            {
                all += elements[j%size];
                sum.Add(all);
            }
        }
        
        return sum.Count;
    }
}